posix/posix.c

```diff
@@ -65,6 +65,10 @@ struct {
 } posix_common;
 
 
+static char hostname[HOST_NAME_MAX];
```

Comment:
You need one more byte for NULL terminator

Reply:
It's intended, hostname_len keeps hostname length.